File: packages/app/src/services/software/definitions/dbBrowserForSqlite.ts
import { ok, err } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import { GithubReleasesScraper } from '../githubReleasesScraper';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'db-browser-for-sqlite';
	public name = 'DB Browser for SQLite';
	public category = [SoftwareCategory.Development, SoftwareCategory.Utility];
	public downloadName = 'DB.Browser.for.SQLite-win64.msi';
	public isArchive = false;
	public shouldCacheUrl = true;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'db-browser-for-sqlite.png';
	public homepage = 'https://sqlitebrowser.org';

	public async resolveDownloadUrl() {
		const owner   = 'sqlitebrowser';
		const repo    = 'sqlitebrowser';
		const release = await GithubReleasesScraper.getRelease(owner, repo, r => r.endsWith('-win64.msi'));
		if (release.isErr()) {
			return err(release.error);
		}

		return ok(release.value);
	}
}

File: src/lib/software/db_browser_for_sqlite.py
from PySide6.QtCore import Slot
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.github_release_scraper import GithubReleaseScraper

class DBBrowserForSQLite(BaseSoftware):
    def __init__(self):
        super().__init__()

        self._gh = GithubReleaseScraper('sqlitebrowser', 'sqlitebrowser', self)
        self._gh.releases_scraped.connect(self._on_releases_scraped)

        self.key = 'db-browser-for-sqlite'
        self.name = 'DB Browser for SQLite'
        self.category = [SoftwareCategory.Development, SoftwareCategory.Utility]
        self.download_name = 'DB.Browser.for.SQLite-win64.msi'
        self.should_cache_url = True
        self.icon = 'db-browser-for-sqlite.png'
        self.homepage = 'https://sqlitebrowser.org'

    @Slot(list)
    def _on_releases_scraped(self, releases: list[str]):
        asset = next((release for release in releases if release.endswith('-win64.msi')), None)
        if asset:
            self.url_resolved.emit(asset)
        else:
            self.url_resolve_error.emit(self.ResolveError.GitHubAssetNotFoundError)

    def resolve_download_url(self):
        self._gh.get_repo_releases()
